src/tokens.rs

```diff
@@ -45,6 +45,12 @@ pub enum Token<T> {
     /// */
     /// ```
     Comment(Comment<T>),
+    /// A hashbang comment, the associated value contains the value.
+    /// ```js
+    /// #!/usr/bin/env node
+    /// process.exit(1)
+    /// ```
+    HashbangComment(Comment<T>),
```

Comment:
I really would prefer if this was a new entry on Comment, the `HTMLComment` is also only allowed 1 line but has it's own entry there.

A big motivator here is that the parser can more easily throw away these comments.

Reply:
Initially I did this specifically to make it easier to handle them specially, but on second thought that doesn't really make sense. will update